crates/bevy_ecs/src/schedule/mod.rs

```diff
@@ -203,320 +186,358 @@ pub fn clear_trackers_system(world: &mut World, resources: &mut Resources) {
+    fn id(&self) -> SystemId {
+        self.system_id
     }
 
-    #[test]
-    fn schedule() {
-        let mut world = World::new();
-        let mut resources = Resources::default();
-        resources.insert(ComputeTaskPool(TaskPool::default()));
-        resources.insert(CompletedSystems::default());
-        resources.insert(1.0f64);
-        resources.insert(2isize);
+    fn update_access(&mut self, _world: &World) {}
 
-        world.spawn((1.0f32,));
-        world.spawn((1u32, 1u64));
-        world.spawn((2u32,));
+    fn archetype_component_access(&self) -> &TypeAccess<ArchetypeComponent> {
+        &self.archetype_component_access
+    }
 
-        let mut stage_a = SystemStage::parallel(); // component queries
-        let mut stage_b = SystemStage::parallel(); // thread local
-        let mut stage_c = SystemStage::parallel(); // resources
+    fn component_access(&self) -> &TypeAccess<TypeId> {
+        &self.component_access
+    }
 
-        // A system names
-        const READ_U32_SYSTEM_NAME: &str = "read_u32";
-        const WRITE_FLOAT_SYSTEM_NAME: &str = "write_float";
-        const READ_U32_WRITE_U64_SYSTEM_NAME: &str = "read_u32_write_u64";
-        const READ_U64_SYSTEM_NAME: &str = "read_u64";
+    fn resource_access(&self) -> &TypeAccess<TypeId> {
+        &self.resource_access
+    }
 
-        // B system names
-        const WRITE_U64_SYSTEM_NAME: &str = "write_u64";
-        const THREAD_LOCAL_SYSTEM_SYSTEM_NAME: &str = "thread_local_system";
-        const WRITE_F32_SYSTEM_NAME: &str = "write_f32";
+    fn is_non_send(&self) -> bool {
+        false
+    }
 
-        // C system names
-        const READ_F64_RES_SYSTEM_NAME: &str = "read_f64_res";
-        const READ_ISIZE_RES_SYSTEM_NAME: &str = "read_isize_res";
-        const READ_ISIZE_WRITE_F64_RES_SYSTEM_NAME: &str = "read_isize_write_f64_res";
-        const WRITE_F64_RES_SYSTEM_NAME: &str = "write_f64_res";
+    unsafe fn run_unsafe(
+        &mut self,
+        _input: Self::In,
+        _world: &World,
+        _resources: &Resources,
+    ) -> Option<Self::Out> {
+        Some(if self.ran {
+            ShouldRun::No
+        } else {
+            self.ran = true;
+            ShouldRun::Yes
+        })
+    }
 
-        // A systems
+    fn apply_buffers(&mut self, _world: &mut World, _resources: &mut Resources) {}
 
-        fn read_u32(completed_systems: Res<CompletedSystems>, _query: Query<&u32>) {
-            let mut completed_systems = completed_systems.completed_systems.lock();
-            completed_systems.insert(READ_U32_SYSTEM_NAME);
-        }
+    fn initialize(&mut self, _world: &mut World, _resources: &mut Resources) {}
+}
 
-        fn write_float(completed_systems: Res<CompletedSystems>, _query: Query<&f32>) {
-            let mut completed_systems = completed_systems.completed_systems.lock();
-            completed_systems.insert(WRITE_FLOAT_SYSTEM_NAME);
-        }
+// TODO more relevant tests
```

Comment:
I think to an extent this is covered by the `stage.rs` tests (order of operations enforcement) and `executor_parallel.rs` tests (testing that conflicting systems are scheduled individually). Can you think of any other tests we should include here or can we remove this?

Reply:
I think I've just missed this one. I wanted to go over it again in search for more cases, but it does look like they're already covered.

Reply:
Removed in f7a0eb63b3e850b8ab93eaa9346a2d085b39ac49